### PR TITLE
Add uk lang сompatibility with version 1.8

### PR DESCRIPTION
**Pull request by iZonex**
Introducing Ukrainian Language Support for Version 1.8

Screenshot 2023-11-17 at 17 37 59
Screenshot 2023-11-17 at 17 12 26

### Changes Description:
- Implemented complete interface localization in Ukrainian.
- Ensured compatibility with the new version 1.8, including interface updates and enhanced functionalities.
- Modified `resources/web/data/text.js` for textual changes in the interface.
- Updated and added localization files `BambuStudio_uk.po` and `.mo` to support the Ukrainian language.

### Purpose of Changes:
The purpose of these changes is to extend the application's accessibility to Ukrainian-speaking users and update compatibility with the latest software version, thereby improving overall performance and user experience.

### CODE REVIEW

**Review by kiddden — APPROVED** [review]
the localization seems to be clear and correct

File: bbl/i18n/uk/BambuStudio_uk.po [hunk 1]
```diff
@@ -3391,7 +3404,7 @@ msgid "Slice plate"
 msgstr "Нарізати моделі"
 
 msgid "Print plate"
-msgstr "Друкована пластина"
+msgstr "Друкувати пластину"
```
Comment (man55) [inline bbl/i18n/uk/BambuStudio_uk.po:3407]: Друкувати стіл

File: bbl/i18n/uk/BambuStudio_uk.po [hunk 2]
```diff
@@ -5468,7 +5511,7 @@ msgid "By Object"
 msgstr "За об'єктом"
 
 msgid "Plate name"
-msgstr "Назва плати"
+msgstr "Назва пластини"
```
Comment (man55) [inline bbl/i18n/uk/BambuStudio_uk.po:5514]: Назва столу

File: bbl/i18n/uk/BambuStudio_uk.po [hunk 3]
```diff
@@ -5613,22 +5656,22 @@ msgid "Busy"
 msgstr "Зайнятий"
 
 msgid "Bambu Cool Plate"
-msgstr "Bambu Cool Plate"
+msgstr "Bambu Cool Пластина"
```
Comment (man55) [inline bbl/i18n/uk/BambuStudio_uk.po:5659]: Cool Plate

File: bbl/i18n/uk/BambuStudio_uk.po [hunk 4]
```diff
@@ -5613,22 +5656,22 @@ msgid "Busy"
...
 
 msgid "PLA Plate"
-msgstr "PLA Plate"
+msgstr "PLA Пластина"
```
Comment (man55) [inline bbl/i18n/uk/BambuStudio_uk.po:5662]: PLA Plate

File: bbl/i18n/uk/BambuStudio_uk.po [hunk 5]
```diff
@@ -5613,22 +5656,22 @@ msgid "Busy"
...
 
 msgid "Bamabu Engineering Plate"
-msgstr "Bamabu Engineering Plate"
+msgstr "Bamabu Engineering Пластина"
```
Comment (man55) [inline bbl/i18n/uk/BambuStudio_uk.po:5665]: Engineering Plate

File: bbl/i18n/uk/BambuStudio_uk.po [hunk 6]
```diff
@@ -5613,22 +5656,22 @@ msgid "Busy"
...
 
 msgid "Bamabu Smooth PEI Plate"
-msgstr ""
+msgstr "Bamabu Smooth PEI Пластина"
```
Comment (man55) [inline bbl/i18n/uk/BambuStudio_uk.po:5668]: Smooth PEI Plate

File: bbl/i18n/uk/BambuStudio_uk.po [hunk 7]
```diff
@@ -5613,22 +5656,22 @@ msgid "Busy"
...
 
 msgid "High temperature Plate"
-msgstr ""
+msgstr "High temperature Пластина"
```
Comment (man55) [inline bbl/i18n/uk/BambuStudio_uk.po:5671]: High temperature Plate

File: bbl/i18n/uk/BambuStudio_uk.po [hunk 8]
```diff
@@ -5613,22 +5656,22 @@ msgid "Busy"
...
 
 msgid "Bamabu Textured PEI Plate"
-msgstr ""
+msgstr "Bamabu Textured PEI Пластина"
```
Comment (man55) [inline bbl/i18n/uk/BambuStudio_uk.po:5674]: Textured PEI Plate

File: bbl/i18n/uk/BambuStudio_uk.po [hunk 9]
```diff
@@ -5829,7 +5877,7 @@ msgid ""
 "Caution to use! Flow calibration on Textured PEI Plate may fail due to the "
 "scattered surface."
 msgstr ""
-"Увага! Калібрування потоку на текстурованій PEI-платі може не вдалося через "
+"Увага! Калібрування потоку на текстурованій PEI-пластини може не вдалося через "
```
Comment (man55) [inline bbl/i18n/uk/BambuStudio_uk.po:5880]: Увага! Калібрування потоку на текстурованій PEI-пластині може не вдастися через

File: bbl/i18n/uk/BambuStudio_uk.po [hunk 10]
```diff
@@ -6180,44 +6228,46 @@ msgid "Nozzle temperature when printing"
 msgstr "Температура сопла під час друку"
 
 msgid "Cool Plate / PLA Plate"
-msgstr "Cool Plate / PLA Plate"
+msgstr "Cool Пластина / PLA Пластина"
```
Comment (man55) [inline bbl/i18n/uk/BambuStudio_uk.po:6231]: Cool Plate / PLA Plate

File: bbl/i18n/uk/BambuStudio_uk.po [hunk 11]
```diff
@@ -6180,44 +6228,46 @@ msgid "Nozzle temperature when printing"
...
 msgstr ""
-"Температура шару при встановленій cool plate. Значення 0 означає, що "
-"філамент\n"
-"не підтримує друк на Cool Plate"
+"Температура шару при встановленій cool plate. Значення 0 означає, що філамент\n"
+"не підтримує друк на Cool Пластині"
 
 msgid "Engineering Plate"
-msgstr "Engineering Plate"
+msgstr "Engineering Пластина"
```
Comment (man55) [inline bbl/i18n/uk/BambuStudio_uk.po:6241]: Engineering Plate

File: bbl/i18n/uk/BambuStudio_uk.po [hunk 12]
```diff
@@ -6180,44 +6228,46 @@ msgid "Nozzle temperature when printing"
...
 "Bed temperature when engineering plate is installed. Value 0 means the "
 "filament does not support to print on the Engineering Plate"
 msgstr ""
-"Температура стола при встановленій Engineering Plate. Значення 0 означає \n"
+"Температура стола при встановленій Engineering Пластина. Значення 0 означає \n"
 "Філамент не підтримує друк на інженерній пластині"
 
 msgid "Smooth PEI Plate / High Temp Plate"
-msgstr ""
+msgstr "Smooth PEI Пластина / High Temp Пластина"
```
Comment (man55) [inline bbl/i18n/uk/BambuStudio_uk.po:6251]: Smooth PEI Plate / High Temp Plate

File: bbl/i18n/uk/BambuStudio_uk.po [hunk 13]
```diff
@@ -6180,44 +6228,46 @@ msgid "Nozzle temperature when printing"
...
 "Value 0 means the filament does not support to print on the Smooth PEI Plate/"
 "High Temp Plate"
 msgstr ""
+"Температура столу, коли встановлена Smooth PEI/High temperature пластина. "
+"Значення 0 означає, що філамент не підтримує друк на Smooth PEI/High "
+"temperature пластині"
 
 msgid "Textured PEI Plate"
-msgstr "Textured PEI Plate"
+msgstr "Textured PEI Пластина"
```
Comment (man55) [inline bbl/i18n/uk/BambuStudio_uk.po:6263]: Textured PEI Plate

File: bbl/i18n/uk/BambuStudio_uk.po [hunk 14]
```diff
@@ -7896,15 +7959,14 @@ msgid ""
 "Close all cooling fan for the first certain layers. Cooling fan of the first "
 "layer used to be closed to get better build plate adhesion"
 msgstr ""
-"Закрийте всі вентилятори, що охолоджують, для перших певних шарів. "
-"Вентилятор охолодження першого шару був закритий, щоб покращити "
-"адгезіюРобочої пластини"
+"Закрийте всі вентилятори, що охолоджують, для перших певних шарів. Вентилятор "
+"охолодження першого шару був закритий, щоб покращити адгезіюРобочої пластини"
```
Comment (man55) [inline bbl/i18n/uk/BambuStudio_uk.po:7963]: охолодження першого шару був закритий, щоб покращити адгезію Робочої пластини

File: bbl/i18n/uk/BambuStudio_uk.po [hunk 15]
```diff
@@ -7896,15 +7959,14 @@ msgid ""
...
 
 msgid "layers"
 msgstr "шари"
 
 msgid "Don't support bridges"
-msgstr "Не підтримувати мости"
+msgstr "Мости без опор"
```
Comment (man55) [inline bbl/i18n/uk/BambuStudio_uk.po:7969]: Мости без підтримок

File: bbl/i18n/uk/BambuStudio_uk.po [hunk 16]
```diff
@@ -11485,8 +11638,7 @@ msgid ""
 msgstr ""
 "Розділіть свої друки на пластини\n"
```
Comment (man55) [inline bbl/i18n/uk/BambuStudio_uk.po:11639]: Розділіть свої друки на столи\n

File: bbl/i18n/uk/BambuStudio_uk.po [hunk 17]
```diff
@@ -11485,8 +11638,7 @@ msgid ""
...
 "Чи знали ви, що ви можете розбивати модель з великою кількістю деталей на "
-"окремі пластини, готові до друку? Це спростить процес відстеження всіх "
-"частин."
+"окремі пластини, готові до друку? Це спростить процес відстеження всіх частин."
```
Comment (man55) [inline bbl/i18n/uk/BambuStudio_uk.po:11641]: окремі столи, готові до друку? Це спростить процес відстеження всіх частин.

**Review by man55 — CHANGES_REQUESTED** [review]
грац

File: bbl/i18n/uk/BambuStudio_uk.po (hunk 4 above)
Comment (iZonex) [reply bbl/i18n/uk/BambuStudio_uk.po:5662]: Please learn English

File: bbl/i18n/uk/BambuStudio_uk.po (hunk 5 above)
Comment (iZonex) [reply bbl/i18n/uk/BambuStudio_uk.po:5665]: Please learn English

File: bbl/i18n/uk/BambuStudio_uk.po (hunk 6 above)
Comment (iZonex) [reply bbl/i18n/uk/BambuStudio_uk.po:5668]: Please learn English

File: bbl/i18n/uk/BambuStudio_uk.po (hunk 13 above)
Comment (iZonex) [reply bbl/i18n/uk/BambuStudio_uk.po:6263]: Please learn English

File: bbl/i18n/uk/BambuStudio_uk.po (hunk 1 above)
Comment (iZonex) [reply bbl/i18n/uk/BambuStudio_uk.po:3407]: Стіл == Table!

File: bbl/i18n/uk/BambuStudio_uk.po (hunk 2 above)
Comment (iZonex) [reply bbl/i18n/uk/BambuStudio_uk.po:5514]: Plate != стол, Plate == пластина

File: bbl/i18n/uk/BambuStudio_uk.po (hunk 15 above)
Comment (iZonex) [reply bbl/i18n/uk/BambuStudio_uk.po:7969]: Please, use to speak international language. "Мости без опор" almost same as "Мости без підтримок"

File: bbl/i18n/uk/BambuStudio_uk.po (hunk 17 above)
Comment (iZonex) [reply bbl/i18n/uk/BambuStudio_uk.po:11641]: same meaning

File: bbl/i18n/uk/BambuStudio_uk.po (hunk 16 above)
Comment (iZonex) [reply bbl/i18n/uk/BambuStudio_uk.po:11639]: Plate != Table

File: bbl/i18n/uk/BambuStudio_uk.po (hunk 14 above)
Comment (iZonex) [reply bbl/i18n/uk/BambuStudio_uk.po]: Fixed thanks!

**Comment by iZonex** [comment]
@lanewei120 Can you please review and merge before release 1.8. We with our Ukraine community did to much improvements for better translation.

**Comment by lanewei120** [comment]
@iZonex 

thanks for doing this
I have just synced the codes of master to newest
could you rebase to the master, and submit it again

we will merge as soon as you finish

**Comment by lanewei120** [comment]
@iZonex 
is it ready now?

**Review by man55 — APPROVED** [review]

**Comment by iZonex** [comment]
> @iZonex is it ready now?

Yes! 

**msgid "Delete Preset"**
msgstr ""

msgid "Are you sure to delete the selected preset?"
msgstr ""

**msgid "Delete preset"**
msgstr ""

**Comment by lanewei120** [comment]
> > @iZonex is it ready now?
> 
> @lanewei120 Yes!
> 
> Please take a look on translations around code base with "Delete Presets" is duplicated is ok, but I think Delete **P**resets and Delete **p**resets is can be duplicate by typo
> 
> **msgid "Delete Preset"** msgstr ""
> 
> msgid "Are you sure to delete the selected preset?" msgstr ""
> 
> **msgid "Delete preset"** msgstr ""

that is ok currently
thanks for doing this

**Comment by lanewei120** [comment]
image

**Comment by lanewei120** [comment]
seems can not rebase
please rebase it manually and submit again

@iZonex

**Comment by iZonex** [comment]
> seems can not rebase please rebase it manually and submit again
> 
> @iZonex

Rebased to one commit for top.

**Pull request merged**